ontotext-yasgui-web-component/src/plugins/yasr/charts-plugin.ts

```diff
@@ -0,0 +1,292 @@
+        case 'http://www.w3.org/2001/XMLSchema#gMonthDay':
+        case 'http://www.w3.org/2001/XMLSchema#gDay':
+        case 'http://www.w3.org/2001/XMLSchema#gMonth':
+          return "number";
+        case 'http://www.w3.org/2001/XMLSchema#date':
+          return "date";
+        case 'http://www.w3.org/2001/XMLSchema#dateTime':
+          return "datetime";
+        case 'http://www.w3.org/2001/XMLSchema#time':
+          return "timeofday";
+        default:
+          return "string";
+      }
+    } else {
+      return "string";
+    }
+  }
+
+  private getGoogleTypeForBindings(bindings, varName) {
+    const types = {};
+    let typeCount = 0;
+    bindings.forEach((binding) => {
+      const type = this.getGoogleTypeForBinding(binding[varName]);
+      if (type != null) {
+        if (!(type in types)) {
+          types[type] = 0;
+          typeCount++;
+        }
+        types[type]++;
+      }
+    });
+    if (typeCount == 0) {
+      return 'string';
+    } else if (typeCount == 1) {
+      for (let type in types) {
+        //just return this one
+        return type;
+      }
+    } else {
+      // we have conflicting types. Throw error
+      console.log('Mapping bindings to types failed', types, varName);
+      throw new TypesMappingError('Mapping bindings to types failed', types, varName);
+    }
+  }
+
+  private castGoogleType(binding, prefixes, googleType) {
+    if (binding == null) {
+      return null;
+    }
+
+    if (googleType != 'string' && binding.type != null && (binding.type === 'typed-literal' || binding.type === 'literal')) {
+      switch (binding.datatype) {
+        case 'http://www.w3.org/2001/XMLSchema#float':
+        case 'http://www.w3.org/2001/XMLSchema#decimal':
+        case 'http://www.w3.org/2001/XMLSchema#int':
+        case 'http://www.w3.org/2001/XMLSchema#integer':
+        case 'http://www.w3.org/2001/XMLSchema#long':
+        case 'http://www.w3.org/2001/XMLSchema#gYearMonth':
+        case 'http://www.w3.org/2001/XMLSchema#gYear':
+        case 'http://www.w3.org/2001/XMLSchema#gMonthDay':
+        case 'http://www.w3.org/2001/XMLSchema#gDay':
+        case 'http://www.w3.org/2001/XMLSchema#gMonth':
+          return Number(binding.value);
+        case 'http://www.w3.org/2001/XMLSchema#date':
+          // the date function does not parse -any- date (including most xsd dates!)
+          // datetime and time seem to be fine though.
+          // so, first try our custom parser. if that does not work, try the regular date parser anyway
+          const date = this.parseXmlSchemaDate(binding.value);
+          if (date) return date;
+        case 'http://www.w3.org/2001/XMLSchema#dateTime':
+        case 'http://www.w3.org/2001/XMLSchema#time':
+          return new Date(binding.value);
+        default:
+          return binding.value;
+      }
+    } else {
+      if (binding.type == 'uri') {
+        return this.uriToPrefixed(prefixes, binding.value);
```

Comment:
You can use "uriToPrefixWithLocalName" function of SparqlUtils

Reply:
Actually I don't want to use it as it's inefissient. I did prefixes to namespace remaping to avoid the redundant looping through the namespaces.